azure_li_services/units/cleanup.py

```diff
@@ -30,15 +32,32 @@ def main():
     service_reports = Defaults.get_service_reports()
 
     reboot_system = False
+    all_services_successful = True
     for report in service_reports:
         if not report.get_state():
             # in case a service has unknown or failed state we will
             # not consider to reboot the machine
+            all_services_successful = False
             reboot_system = False
             break
         if report.get_reboot():
             reboot_system = True
 
+    install_source = Defaults.mount_config_source()
+    try:
+        state_file = os.sep.join(
+            [
+                install_source.location,
+                'workload_success_is_{}'.format(
+                    all_services_successful
+                ).lower()
+            ]
+        )
+        with open(state_file, 'w'):
+            pass
+    finally:
+        Defaults.umount_config_source(install_source)
+
```

Comment:
So this is part of what I was hoping for, but not all. This will show me that the setup process succeeded or failed, but it won't tell me what was wrong.

I suppose if a failure exists, I could then to go the console to get an idea of what happened. But my hope was that I'd get that information and more, particularly if things failed.

Remember, we don't generally have login credentials to these systems except under test. So if a customer deployment failed (customer would generally give us a public certificate for login), we couldn't even log in to see what failed and why. So my hope: if `workload_success_is_false` was created, I was hoping it would contain enough information so we could see what failed - and why. Otherwise, we'd need to try and redeploy with our own credentials, hopefully replicating the error, resolving it, then redeploying for the customer.

Reply:
There's a range of reasons you may not be able to log in, all unrelated to security aspects and trust. For example, bad YAML, but YAML version, bug causing a crash in account setup ... these are all reasons that would create a scenario where you can't log in. But if you can't log in, you can't see exactly what the problem is (best you can do is interpolate the failures based on what services failed overall).

In general, I've never seen sensitive data in the logs from `systemctl`, although I suppose it's possible. Keep in mind, however, that access to the configuration LUN is limited to:

* The machine that is booting, and
* The management VLAN.

Customers should never be able to get access to the overall LUN information, even customers that own the machine in question. Automation will unmap the LUN once setup is complete (thus our reasoning for asking for a way to know when setup is complete).

I'm totally fine discussing this in another issue, which you already opened. So I'll sign off on this and discuss there.

Reply:
ok if you don't see a potential problem with instance data put on the LUN I'll be fine with it too